app/src/dev/java/com/mycompany/myapp/app/VariantSettings.java

```diff
@@ -0,0 +1,40 @@
+package com.mycompany.myapp.app;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import com.mycompany.myapp.R;
+
+public class VariantSettings {
+    private static final String PREFS_SETTINGS = "settings";
+
+    private static final String PREF_BASE_URL = "base_url";
+    private static final String PREF_TRUST_ALL_SSL = "trust_all_ssl";
+
+    private final Context context;
+
+    public VariantSettings(Context context) {
+        this.context = context;
+    }
+
+    public String getBaseUrl() {
+        String defaultBaseUrl = context.getString(R.string.default_base_url);
+        return getSettingsSharedPreferences().getString(PREF_BASE_URL, defaultBaseUrl);
+    }
+
+    public void setBaseUrl(String baseUrl) {
+        getSettingsSharedPreferences().edit().putString(PREF_BASE_URL, baseUrl).apply();
```

Comment:
I like the variant settings setup here.  Definitely a useful option for any Android project.